### ISSUE TITLE
bindP Nothing - variable not bound

### ISSUE THREAD

**Issue by bagl** [paulrzcz/takusen-oracle#8]
Currently it seems `bindP Nothing` does not bind the parameter as I receive **ORA-01008: not all variables bound** when calling `execDDL $ cmdbind $ procSQL listOfBinds`

Why not binding the parameter to sql's `NULL` value?

**Comment by paulrzcz** [comment]
Sorry for the extremely long reply.

Having studied NULL handing in the library, the library seems to bind Nothing to NULL by means of indicator variable in OCIBindByPos foreign function.

Could you, please, provide more elaborate example? It is interesting to see into procSQL and listOfBinds.

**Comment by bagl** [comment]
Sorry for the late reply, too...

Here is procedure in Oracle:

```plsql
PROCEDURE test1 (num_ number)
AS
BEGIN
  insert into test_table(num) values (num_);
END;
```
and the way I am calling it from Haskell:

```haskell
flip catchDB reportRethrow
  $ withSession (connect "user" "pass" "server")
  $ execDDL $ cmdbind "begin test1(?); commit; end;" [ bindP (Nothing :: Maybe Int) ]
```
which crashes with *ORA-01008: not all variables bound*, however it works as expected when I use `[ bindP (Just 1 :: Maybe Int) ]`.

**Comment by bagl** [comment]
OK, I've learned some C and looked into it and I have a suspect.

[bindWithValue](https://github.com/paulrzcz/takusen-oracle/blob/0706e52e306062024c091d0e055ecac66c99220c/Database/Oracle/Enumerator.hs#L712-L714) for `Nothing` case is `return ()`. So it does nothing!

The indicator value is set to `-1` as it [should be](http://docs.oracle.com/cd/B28359_01/appdev.111/b28395/oci02bas.htm#LNOCI87631). So now we need to bind something (anything? - it should be ignored according to OCI docs) as a value. I tried to bind a `nullPtr` 

```haskell
bindWithValue Nothing a = a nullPtr -- instead of return ()
```

and it works, but I do not know what the consequences are, so help still needed :)
